### PR TITLE
[ON-2881] Fix delete inventory

### PR DESCRIPTION
**Pull request by thehighestprimenumber**
### What change is being made?
Add functionality to delete an inventory through the DeleteInventoryModal component with password confirmation.

### Why are these changes being made?
The feature update resolves the need for secure inventory deletion within the application by integrating API requests for password verification and deletion, and by providing user feedback upon successful or unsuccessful attempts. This enhances the user experience by ensuring necessary security protocols are followed and confirmations are communicated effectively.

> Is this description stale? Ask me to generate a new description by commenting `/korbit-generate-pr-description`

### CODE REVIEW

File: app/src/services/api.ts [hunk 1]
```diff
@@ -48,6 +49,7 @@ export const api = createApi({
     "SectorBreakdown",
     "Inventory",
     "CitiesAndInventories",
+    "Inventories",
```
Comment (lemilonkh) [inline app/src/services/api.ts:52]: Can we use the Inventory tag we have already here? Normally we don't distinguish between singular/ plural in these tags...

File: app/src/services/api.ts [hunk 2]
```diff
@@ -289,6 +291,18 @@ export const api = createApi({
...
+      InventoryAttributes,
+      InventoryDeleteQuery
+    >({
+      query: (data) => ({
+        url: `/inventory/${data.inventoryId}`,
+        method: "DELETE",
+      }),
+      transformResponse: (response: { data: InventoryAttributes }) =>
+        response.data,
+      invalidatesTags: ["InventoryProgress", "InventoryValue", "Inventories"],
```
Comment (lemilonkh) [inline app/src/services/api.ts:304]: Technically this also invalidates Inventory, ActivityValue, InventoryValue, ReportResults, YearlyReportResults...

Probably won't cause issues for now as you will be on another inventory page or onboarding afterwards but still 😅

**Review by lemilonkh — APPROVED** [review]
Thanks 👍

File: app/src/services/api.ts (hunk 1 above)
Comment (thehighestprimenumber) [reply app/src/services/api.ts:52]: I had to add the Inventories so that the list of inventories gets updated without the inventory you just deleted

**Pull request merged**